### ISSUE TITLE
parser: incorrect `export_kind` of `export type {} from 'mod'`

### ISSUE THREAD

**Issue by Dunqing** [oxc-project/oxc#10465]
Regression probably caused by https://github.com/oxc-project/oxc/pull/10389

```ts
export type {} from 'mod'
```

In this case, the `export_kind` should be `type`, but it is `value` now. See [Oxc Playground](https://playground.oxc.rs/#eNo9j7tuwzAMRX9F4JLF6Avo4q5B1hbt2oWW6UCATAoUlcYw/O+VY9gTHyDPvXcGDy3QPYmasymRm93ZLW5QGd3p6bk/ffwyNCDQzqCF15InNrxDa1qogRjY9j57SXQM09hJ3CdT5DyIjtAOGDMtDSTUTLoSMUb5+yYryp/FcujpUthbkKq3vSelenujL1TivG0rYhV/IGp/KGxMQ71SdQaU315e32uGkCWiUX8mH1FxxVfS7sZLT1d6BCTGLtKPFPU0YjrUxsBhCLueFzaVeKnW16cbaSe5pt+Ay/IPDj57dg==) and switch to `Rust AST` Tab. But this is correct in `EStree AST`.

@leaysgur cc

**Comment by leaysgur** [comment]
Sorry && thanks for letting me know!

It will be tomorrow, but I will check it out.

**Comment by camc314** [comment]
@Dunqing is the issue that given the following:

```
export type { Foo } from 'mod'
```

`Foo` currently has export_kind of value, but it should be type?

[typescript-eslint has `value` for the specifier export kind](https://typescript-eslint.io/play/#ts=5.8.2&showAST=es&fileType=.tsx&code=KYDwDg9gTgLgBDAnmYcDeAoAZhCGC%2BcWUEAtnAOQB0FA3EA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

**Comment by Boshen** [comment]
I cannot reproduce the original issue.

**Comment by Dunqing** [comment]
It's still incorrect somewhere, see [Oxc playground](https://playground.oxc.rs/#eNo9j7tuwzAMRX9F4JLF6Avo4q5B1hbt2oWW6UCATAoUlcYw/O+VY9gTHyDPvXcGDy3QPYmasymRm93ZLW5QGd3p6bk/ffwyNCDQzqCF15InNrxDa1qogRjY9j57SXQM09hJ3CdT5DyIjtAOGDMtDSTUTLoSMUb5+yYryp/FcujpUthbkKq3vSelenujL1TivG0rYhV/IGp/KGxMQ71SdQaU315e32uGkCWiUX8mH1FxxVfS7sZLT1d6BCTGLtKPFPU0YjrUxsBhCLueFzaVeKnW16cbaSe5pt+Ay/IPDj57dg==), and switch tab to `Codegen`, it prints `export { D } from './d'`, but it shouldn't be `export type { D } from './d'`?

**Comment by overlookmotel** [comment]
Fixed in #10539.
